server/storage/couchdb.go

```diff
@@ -0,0 +1,591 @@
+package storage
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/theupdateframework/notary/storage/couchdb"
+	"github.com/theupdateframework/notary/tuf/data"
+
+	"github.com/flimzy/kivik"
+)
+
+// CDBTUFFile is a TUF file record
+type CDBTUFFile struct {
+	couchdb.ID
+	couchdb.Timing
+	GunRoleVersion []interface{} `json:"gun_role_version"`
+	Gun            string        `json:"gun"`
+	Role           string        `json:"role"`
+	Version        int           `json:"version"`
+	SHA256         string        `json:"sha256"`
+	Data           []byte        `json:"data"`
+	TSchecksum     string        `json:"timestamp_checksum"`
+}
+
+// TableName returns the table name for the record type
+func (r CDBTUFFile) TableName() string {
+	return TUFFileTableName
+}
+
+// CChange defines the the fields required for an object in the changefeed
+type CChange struct {
+	couchdb.ID
+	CreatedAt time.Time `json:"created_at"`
+	GUN       string    `json:"gun"`
+	Version   int       `json:"version"`
+	SHA256    string    `json:"sha256"`
+	Category  string    `json:"category"`
+}
+
+// TableName sets a specific table name for Changefeed
+func (cdb CChange) TableName() string {
+	return ChangefeedTableName
+}
+
+// unmarshal in an anonymous struct
```

Comment:
Correct. It looks like it can be removed.

```
diff --git a/server/storage/couchdb.go b/server/storage/couchdb.go
index bc86a608..da141ff9 100644
--- a/server/storage/couchdb.go
+++ b/server/storage/couchdb.go
@@ -49,38 +49,6 @@ func (cdb CChange) TableName() string {
 	return ChangefeedTableName
 }
 
-// unmarshal in an anonymous struct
-func cdbTUFFileFromJSON(data []byte) (interface{}, error) {
-	a := struct {
-		CreatedAt  time.Time `json:"created_at"`
-		UpdatedAt  time.Time `json:"updated_at"`
-		DeletedAt  time.Time `json:"deleted_at"`
-		Gun        string    `json:"gun"`
-		Role       string    `json:"role"`
-		Version    int       `json:"version"`
-		SHA256     string    `json:"sha256"`
-		Data       []byte    `json:"data"`
-		TSchecksum string    `json:"timestamp_checksum"`
-	}{}
-	if err := json.Unmarshal(data, &a); err != nil {
-		return CDBTUFFile{}, err
-	}
-	return CDBTUFFile{
-		Timing: couchdb.Timing{
-			CreatedAt: a.CreatedAt,
-			UpdatedAt: a.UpdatedAt,
-			DeletedAt: a.DeletedAt,
-		},
-		GunRoleVersion: []interface{}{a.Gun, a.Role, a.Version},
-		Gun:            a.Gun,
-		Role:           a.Role,
-		Version:        a.Version,
-		SHA256:         a.SHA256,
-		Data:           a.Data,
-		TSchecksum:     a.TSchecksum,
-	}, nil
-}
-
 func cdbChangeFromJSON(data []byte) (interface{}, error) {
 	res := CChange{}
 	if err := json.Unmarshal(data, &res); err != nil {
diff --git a/server/storage/couchdb_models.go b/server/storage/couchdb_models.go
index fd2a2b7e..5d98fc74 100644
--- a/server/storage/couchdb_models.go
+++ b/server/storage/couchdb_models.go
@@ -17,7 +17,6 @@ var (
 			"6": []string{"gun", "created_at"},
 			"7": []string{"_id", "created_at"},
 		},
-		JSONUnmarshaller: cdbTUFFileFromJSON,
 	}
 
 	// ChangeCouchTable is the table definition for changefeed objects

```